zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/serializer/LongSerializerTest.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.serializer;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonStreamContext;
+import io.camunda.zeebe.gateway.protocol.rest.UserTaskItem;
+import io.camunda.zeebe.gateway.protocol.rest.UserTaskSearchQueryResponse;
+import io.camunda.zeebe.gateway.rest.RequestMapper;
+import io.camunda.zeebe.util.Either.Left;
+import jakarta.servlet.http.HttpServletRequest;
+import java.io.IOException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpHeaders;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+class LongSerializerTest {
```

Comment:
🔧 When we added detection of the Java client, let's also test this here in the following scenarios at least:
1. The accept header is set (e.g. to String) and there is a Java client 8.6 or 8.5 agent header => string keys
2. No accept header is set and there is a Java client 8.6 or 8.5 agent header => number keys
3. No accept header is set and there is a Java client 8.7 header => string keys